BEPs/BEP-336.md

```diff
@@ -0,0 +1,108 @@
+    - [6.5 Fetching of Sidecars when a node starts](#65-fetching-of-sidecars-when-a-node-starts)
+  - [7. Changes Specifc to BSC](#7-changes-specifc-to-bsc)
+    - [7.1 Verification](#71-verification)
+    - [7.2 Sidecar Propagation / P2P network](#72-sidecar-propagation--p2p-network)
+    - [7.3 Sidecar Fetching/Downloading](#73-sidecar-fetchingdownloading)
+  - [8. License](#8-license)  
+
+## 1. Summary
+
+This BEP proposes introduction of blob-carrying transactions which may contain large amount of data.
+
+## 2. Abstract 
+
+Introduce a new transaction format for “blob-carrying transactions” which contain a large amount of data that cannot be accessed by EVM execution, but whose commitment can be accessed. The format is intended to be fully compatible with the format that will be used in full sharding.
+
+## 3. Status 
+
+Work in progress. 
+A PoC has been created in this branch: https://github.com/bnb-chain/bsc/tree/4844 
+
+## 4. Motivation 
+
+Rollups such as OpBnB offer scaling solutions of BSC. But their ability might be limited sometimes due to higher transaction fees and cost of storage in the blockchain.
+
+This BEP provides a solution by introducing blob-transaction which carry blobs that can be arbitrary data (e.g. from a rollup) and they act as temporary storage. Therefore they cost less and have the potential to make it cheaper for rollups to interact with BSC.
+
+## 5. Ethereum Specification 
+
+Details of Ethereum specification can be found [here](https://eips.ethereum.org/EIPS/eip-4844). 
+The specification from execution layer will be same in case of BSC. The specifications from consensus layer will differ as BSC doesn't have execution-consensus separation & these will be covered in subsequent sections.
+
+Some key specifications that will be same as Ethereum include:
+
+- Blob data  stored separately from blocks as "sidecars"
+- Polynomial Commitment Scheme & Verification Logic
+- Addition of a blob transaction to Blob pool & corresponding checks
+- Transaction pooling, verification and propagation to support blob transactions and sidecars
+
+## 6. Life of a Blob Transaction
+
+![Life of a blob transaction](./assets/BEP-336/blobtxlife.png)
+
+### 6.1 Blob pool
+Once a blob transaction is sent, it ends up in the blob pool which is a separate pool only for blob transactions.
+
+### 6.2 Verification / Filtering
+The blob transaction undergoes initial verification just like in case of a normal transaction.
+There's however an extra verification step in `validateBlobSidecar()`. This should verify proofs of the blobs using `VerifyBlobProof()`. 
+Once the verification is passed the transaction is stored in the blobpool storage.
+
+### 6.3 Consensus process: Generation & Storage of Sidecars
+During consensus process, the sidecars consisting of blobs are made separately than a block. A block may have multiple sidecars. The payload of execution will include sidecars and the rest will remain the same.
```

Comment:
The payload of execution will include sidecars and the rest will remain the same.
===
Q: what does it mean? Legacy transaction's payload will include sidecars content?

Reply:
I should probably change the wording. 
The point is that during consensus process, the sidecars are needed to be constructed. There are two ways of doing it. 
1 - Include the blobs etc inside the transactions themselves and fetch those during consensus directly from transactions.
2 - Transactions once added to the pool don't have blobs with them. This is what geth does to keep it lightweight. Later during consensus process, prysm separately calls to receive the blobs which are then converted to sidecars. 

In my PoC which I implemented before, I had done the first one as I mainly wanted to get it working as soon as possible. It is unclear to me if the 2nd point is necessarily the best way to go for bsc and if the performance will significantly reduce if we do the first point. 

Once I reach that stage in my current work after merging from geth, I will be in a better position.

Reply:
Blob are propagated differently from Tx:
- blob payload are broadcasted less frequently to reduce the traffic cost
- only recent blob will be kept, around 18days on Ethereum
IMHO, we need the blob pool.